frideos_library/lib/app_state.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:path_provider/path_provider.dart';
+
+import 'package:todos_repository_simple/todos_repository_simple.dart';
+
+import 'package:frideos/frideos.dart';
+
+import 'package:frideos_library/blocs/stats_bloc.dart';
+import 'package:frideos_library/blocs/todos_bloc.dart';
+import 'package:frideos_library/models/models.dart';
+
+class AppState extends AppStateModel {
+  static final AppState _singletonAppState = AppState._internal();
+
+  AppState._internal();
+
+  factory AppState() {
+    return _singletonAppState;
+  }
+
+  TodosRepositoryFlutter respository;
+  TodosBloc todosBloc;
+  StatsBloc statsBloc;
+  final tabController = StreamedValue<AppTab>(initialData: AppTab.todos);
+
+  @override
+  init() {
+    respository = const TodosRepositoryFlutter(
+      fileStorage: const FileStorage(
+        'frideos_library',
+        getApplicationDocumentsDirectory,
+      ),
+    );
+
+    todosBloc = TodosBloc(repository: respository);
+    statsBloc = StatsBloc();
+
+    todosBloc.todosSender.setReceiver(statsBloc.todosItems);
```

Comment:
Just a Q about the pattern, no changes required: Do you think this pattern will scale well to a large app? Instead of "tunneling" a list between two blocs, I generally require my blocs to depend on a common data source (ReactiveTodosRepository in the case of the bloc examples).

My initial impression: As an app grows, you're going to need to manage these receivers pretty carefully and might end up with some "event spaghetti." What are your thoughts?

Reply:
Exemplary question. This pattern was conceived as an easy way to share data between two blocs but not as an alternative to the data layering that I think it is by far the best method to share data between two or more blocs in complex apps. It could be easily used specifically in certain situations where you just need to send some data only between two blocs (e.g. I wrote an example app where the user can select some boxes from a page and send them to another one), and you want a little bit more control over it, by using a specific send method you know exactly how and when you are sending the data just giving a look at the code.

Reply:
Thanks for the explanation!